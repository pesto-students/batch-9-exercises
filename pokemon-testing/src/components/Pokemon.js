import React from 'react';
import PropTypes from 'prop-types';

const Pokemon = ({ pokemon }) => (
  <li className="pokemons__item">
    <button
      type="button"
      className="pokemons__sprite"
      style={{
        backgroundImage: `url(${`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${pokemon.id}.png`})`,
      }}
    />
    <p className="pokemons__name">{pokemon.name}</p>
  </li>
);

Pokemon.propTypes = {
  pokemon: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default Pokemon;
