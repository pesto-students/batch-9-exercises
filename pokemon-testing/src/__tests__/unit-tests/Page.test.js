import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Page from '../../containers/Page/Page';
import Pokemon from '../../components/Pokemon';

describe('<Page />', () => {
  describe('matches the snapshot', () => {
    test('when isFetched is true', () => {
      const fetchPokemons = jest.fn();
      const filterPokemons = jest.fn();
      const props = {
        page: {
          isFetched: true,
          displayedPokemons: [],
        },
        pageActions: {
          fetchPokemons,
          filterPokemons,
        },
      };
      const tree = shallow(<Page {...props} />);
      expect(toJson(tree)).toMatchSnapshot();
    });

    test('when isFetched is false', () => {
      const fetchPokemons = jest.fn();
      const filterPokemons = jest.fn();
      const props = {
        page: {
          isFetched: false,
          displayedPokemons: [{ id: 1, name: 'balbasaur' }],
        },
        pageActions: {
          fetchPokemons,
          filterPokemons,
        },
      };
      const tree = shallow(<Page {...props} />);
      expect(toJson(tree)).toMatchSnapshot();
    });
  });

  test('should have loading text when pokemons are not fetched', () => {
    const fetchPokemons = jest.fn();
    const filterPokemons = jest.fn();
    const props = {
      page: {
        isFetched: true,
        displayedPokemons: [],
      },
      pageActions: {
        fetchPokemons,
        filterPokemons,
      },
    };
    const PageComponent = shallow(<Page {...props} />);
    const loadingTextTag = PageComponent.find('p');
    expect(loadingTextTag.length).toBe(1);
    expect(loadingTextTag.text()).toBe('Loading...');
  });
  test('should have pokemons  when pokemons are fetched', () => {
    const fetchPokemons = jest.fn();
    const filterPokemons = jest.fn();
    const props = {
      page: {
        isFetched: false,
        displayedPokemons: [{ id: 1, name: 'balbasaur' }],
      },
      pageActions: {
        fetchPokemons,
        filterPokemons,
      },
    };
    const PageComponent = shallow(<Page {...props} />);
    const pokemons = PageComponent.find(Pokemon);
    expect(pokemons.length).toBe(1);
  });
  test('should throw when no props are passed', () => {
    expect(() => shallow(<Page />)).toThrow();
  });
});
