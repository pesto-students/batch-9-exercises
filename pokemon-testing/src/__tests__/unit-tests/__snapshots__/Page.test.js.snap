// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> matches the snapshot when isFetched is false 1`] = `
<div
  className="pokemons__wrapper"
>
  <Search
    onChange={[Function]}
  />
  <ul
    className="pokemons"
  >
    <Pokemon
      key="0"
      pokemon={
        Object {
          "id": 1,
          "name": "balbasaur",
        }
      }
    />
  </ul>
</div>
`;

exports[`<Page /> matches the snapshot when isFetched is true 1`] = `
<div
  className="pokemons__wrapper"
>
  <Search
    onChange={[Function]}
  />
  <ul
    className="pokemons"
  >
    <p>
      Loading...
    </p>
  </ul>
</div>
`;
